### PR TITLE
Update 'response_waits' map.

### PR DESCRIPTION
**Pull request by GTwhy**
LockFreeCuckooHash is used to replace Hashmap and WorkRequestID is used to replace RequestID, ensuring the uniqueness of the ID used in each request and avoiding the crash that may be caused by ID duplication.

### CODE REVIEW

File: src/agent.rs [hunk 1]
```diff
@@ -481,16 +480,20 @@ impl AgentInner {
     /// Send a request with data appended
     async fn send_request_append_data(
         &self,
-        request: Request,
+        mut request: Request,
         lm: &[&LocalMemoryRegion],
     ) -> io::Result<ResponseKind> {
-        let (send, recv) = oneshot::channel();
+        let (tx, mut rx) = channel(2);
```
Comment (rogercloud) [inline src/agent.rs:486]: As there're too many kinds of channels, specify that it's oneshot is more clear.

File: src/agent.rs [hunk 2]
```diff
@@ -481,16 +480,20 @@ impl AgentInner {
     /// Send a request with data appended
     async fn send_request_append_data(
         &self,
-        request: Request,
+        mut request: Request,
```
Comment (rogercloud) [inline src/agent.rs:483]: We don't recommend to make parameter mutable, we can add a function to creating a new Request with new WorkRequestId based on the old one.

File: src/completion_queue.rs [hunk 3]
```diff
@@ -205,7 +206,7 @@ impl From<WCError> for io::Error {
 }
 
 /// Work request id
-#[derive(PartialEq, Eq, Hash, Debug, Clone, Copy)]
+#[derive(PartialEq, Eq, Hash, Debug, Clone, Copy, Serialize, Deserialize)]
```
Comment (rogercloud) [inline src/completion_queue.rs:209]: Why do you add these traits?

Comment (GTwhy) [reply src/completion_queue.rs:209]: This was added to enable WorkRequestID to replace RequestID of Agent. But we'll refactor this part to make it more logical to abstract.

File: src/agent.rs (hunk 1 above)
Comment (GTwhy) [reply src/agent.rs:486]: Thanks for the advice, but for now it will all be MPSC, because oneshot does not meet the needs of LockFreeHashMap.

Comment (rogercloud) [reply src/agent.rs:486]: So put the comment for the reason of the channel choice. So the code reader can understand it.

File: src/id.rs [hunk 4]
```diff
@@ -0,0 +1,16 @@
+use rand::Rng;
+use std::time::{SystemTime, UNIX_EPOCH};
+use utilities::OverflowArithmetic;
+
+/// Creat a random u64 id
+pub(crate) fn random_u64() -> u64 {
```
Comment (rogercloud) [inline src/id.rs:6]: Put more meaningful comments here to explain why we design the id this way.

**Review by rogercloud — APPROVED** [review]

**Pull request merged**